waspc/data/Generator/templates/react-app/src/api.ts

```diff
@@ -59,9 +59,12 @@ export const handleApiError = (error) => {
     // TODO: We might want to use HttpError here instead of just Error, since
     //   HttpError is also used on server to throw errors like these.
     //   That would require copying HttpError code to web-app also and using it here.
-    const responseJson = error.response?.data
+    const responseJson = error.response?.data as {
+      message?: string
+      data?: unknown
+    }
     const responseStatusCode = error.response.status
-    const e = new Error(responseJson?.message || error.message)
+    const e: WaspError = new Error(responseJson?.message || error.message)
```

Comment:
When creating custom errors, it's better to extend the `Error` class and use it like this:
```typescript
throw new WaspError(responseJson?.message ?? error.message, responseStatusCode, responseJson?.data)
```